### ISSUE TITLE
Snap installation shows two icon in Gnome dock

### ISSUE THREAD

**Issue by zypA13510** [a85/POSTMan-Chrome-Extension#6920]
**Describe the bug**
Basically the same as https://bugs.launchpad.net/snappy/+bug/1718942
When I add the snap version of postman to favorite, and then open it, two icons show up in the dock, instead of one combined icon (see screenshot below).

According to the bug report, I have made the following modification to `/var/lib/snapd/desktop/applications/postman_postman.desktop`:
```patch
--- /var/lib/snapd/desktop/applications/postman_postman.desktop
+++ postman_modified.desktop
@@ -7,4 +7,4 @@
 Categories=Development;
 MimeType=x-scheme-handler/postman
 Terminal=false
-
+StartupWMClass=Postman
```
and the issue is resolved.

I'm not quite sure if this is a postman issue, a snap issue, or maybe a dash-to-dock issue, if the .desktop file is supplied by postman, I think you can fix it there.

**To Reproduce**
1. Install snap version of postman
    ```console
    $ snap install postman
    ```
2. Add it to favorites in Gnome
3. Run postman

**Expected behavior**
Only one combined icon of postman shows up in the dock.

**Screenshots**
<img width="68" alt="2019-07-15 10_42_31" src="https://user-images.githubusercontent.com/8077540/61193357-904df980-a6ed-11e9-9ded-a7469685bc55.png">

**App information (please complete the following information):**
 - App Type: Snap
 - Postman Version: 6.7.1
 - OS: CentOS 7 (3.10.0-957.21.3.el7.x86_64)

**Additional context**
More environment information (CentOS w/ Gnome + dash-to-dock):
```console
$ yum list gnome-shell-extension-dash-to-dock epel-release snapd
Installed Packages
epel-release.noarch                              7-11                   @extras 
gnome-shell-extension-dash-to-dock.noarch        3.28.1-5.el7.1         @updates
snapd.x86_64                                     2.39.2-1.el7           @epel
$ snap version
snap    2.39.2-1.el7
snapd   2.39.2-1.el7
series  16
centos  7
kernel  3.10.0-957.21.3.el7.x86_64
$ snap list postman
Name     Version  Rev  Tracking  Publisher     Notes
postman  6.7.1    81   stable    postman-inc✓  -
```
Somehow I'm unable to reproduce this on Ubuntu 18.04.2 with ubuntu-dock (I checked and `StartupWMClass` wasn't specified by default):
```console
$ apt list gnome-shell-extension-ubuntu-dock snapd
Listing... Done
gnome-shell-extension-ubuntu-dock/bionic-updates,now 0.9.1ubuntu18.04.3 all [installed]
snapd/bionic-updates,now 2.39.2+18.04 amd64 [installed]
$ snap version
snap    2.39.3
snapd   2.39.3
series  16
ubuntu  18.04
kernel  4.15.0-54-generic
$ snap list postman
Name     Version  Rev  Tracking  Publisher     Notes
postman  6.7.1    81   stable    postman-inc✓  -
```

**Comment by mccannt** [comment]
@zypA13510 Are you still experiencing this issue?

**Comment by zypA13510** [comment]
Yes, it still happens, and the `StartupWMClass` fix above still works.

**Comment by mccannt** [comment]
Thanks for letting us know @zypA13510 
We will investigate this issue further and let you know if there is anything else we need.

**Comment by jviotti** [comment]
Hey there! We have revamped our Snap packaging routines and I believe the issue is gone with later versions of the app. I'll close the issue, but we'll re-open if you find this is still happening on your system on the latest version of the Snap.

**Comment by ryanotella** [comment]
This is still a problem. The work-around still works. :)

**Comment by vageliskatsiotis** [comment]
Issue still exists and not a specific app issue I think. Arch Linux. Workaround still works after using xprop to get the app's WM_CLASS. I would say it's likely snap / gnome issue

**Comment by MC-sanderv** [comment]
This comes down to how GNOME docks handle Snap apps and .desktop metadata rather than a functional issue in Postman itself. 

Since this is environment specific and doesn’t affect app behavior, closing this issue.
Should someone still experience the issue, the above mentioned fix should work if still relevant.
